index.html

```diff
@@ -555,7 +556,102 @@ <h2>Dictionary {{RTCRtpEncodingParameters}} Members</h2>
+          </li>
+        </ol>
+      </li>
+      <li>
+        <p>Else (<var>codecs</var> is an empty list), if <var>transceiver</var>.
+          {{RTCRtpTransceiver/[[PreferredCodecs]]}} is not an empty list:</p>
+        <ol>
+          <li>
+            <p>If any <var>codec</var> parameter in <var>encodings</var> does not match
+              any entry in the <var>transceiver</var>.{{RTCRtpTransceiver/[[PreferredCodecs]]}},
+              return a promise [= rejected =] with a newly [= exception/created =]
+              {{InvalidModificationError}}.</p>
+          </li>
+        </ol>
+      </li>
+      <li>
+        <p>Else ({{RTCRtpTransceiver/[[PreferredCodecs]]}} is an empty list):</p>
+        <ol>
+          <li>
+            <p>If any <var>codec</var> parameter in <var>encodings</var> does not match
+              any codec in {{RTCRtpSender.getCapabilities(kind)}}.<code>codecs</code>,
+              return a promise [= rejected =] with a newly [= exception/created =]
+              {{InvalidModificationError}}.</p></li>
+        </ol>
+      </li>
+      <li>
+        <p>If the user agent does not support setting the codec for any encoding or mixing
+          different codec values on the different encodings, return a promise [= rejected =]
+          with a newly [= exception/created =] {{OperationError}}.
+        </p>
+      </li>
+    </ol>
+    <h3>addTransceiver()</h3>
+    <p>Add the following steps in the <var>sendEncodings</var> validation for
+      {{RTCPeerConnection/addTransceiver()}}, before any validation of
+      <var>scalabilityMode</var> from [[WEBRTC-SVC]]:</p>
+    <ol>
+      <li>
+        <p>If any <var>codec</var> parameter in <var>sendEncodings</var> does not match any codec in
+        {{RTCRtpSender.getCapabilities(kind)}}.<code>codecs</code>,
+        [= exception/throw =] an {{OperationError}}.</p>
+      </li>
+      <li>
+        <p>If the user agent does not support changing the codec for any encoding or mixing
+          different codec values on the different encodings, 
+          [= exception/throw =] an {{OperationError}}.</p>
+      </li>
+    </ol>
+    <h3>Set the session description</h3>
+    <p>Append the following steps to the <a data-cite="WEBRTC#set-description">set the session description</a> algorithm:</p>
+    <ol>
+      <li>
+        <p>For each <var>transceiver</var> in <var>connection</var>'s [= set of transceivers =]:</p>
+        <ol>
+          <li>
+            <p>Let <var>codecs</var> be <var>transceiver</var>.{{RTCRtpTransceiver/[[Sender]]}}.{{RTCRtpSender/[[SendCodecs]]}}.</p>
+          </li>
+          <li>
+            <p>If <var>codecs</var> is not an empty list:</p>
+            <ol>
+              <li>
+                <p>Remove any <var>codec</var> value in <var>transceiver</var>.{{RTCRtpTransceiver/[[Sender]]}}.{{RTCRtpSender/[[SendEncodings]]}}
+                  that does not match any entry in <var>codecs</var>.</p>
```

Comment:
Edge case that needs clarification: If a codec that we have previously preferred (encodings[0].codec = "VP8" for example) is removed in response to negotiation, will encodings[0].codec become missing (no preference) or will we continue to prefer VP8 just in case it is negotiated in the future?

Reply:
The goal is to remove the `codec` value override in each entry in `[[SendEncodings]]` that became invalid after negotiation. Do you have a better wording to suggest?

Reply:
Nevermind, I was reading this wrong. After re-reading, I think this is clear as-is. LGTM without any changes.